File: R/ExtendedEuclidean.r
ExtendedEuclidean <- function(alpha_len, a)
{
  remainder <- c(alpha_len, a)
  alpha_len_s <- c(1, 0)
  a_t <- c(0, 1)
  quotient <- c(0, 0)
  
  while (remainder[length(remainder)] > 0)
  {
    quotient <-
      c(quotient, floor(remainder[length(remainder) - 1] / remainder[length(remainder)]))
    remainder <-
      c(remainder, remainder[length(remainder) - 1] %% remainder[length(remainder)])
    alpha_len_s <-
      c(alpha_len_s, alpha_len_s[length(alpha_len_s) - 1] - quotient[length(quotient)] * alpha_len_s[length(alpha_len_s)])
    a_t <-
      c(a_t, a_t[length(a_t) - 1] - quotient[length(quotient)] * a_t[length(a_t)])
    
  }
  
  
  return(list(gcd = remainder[length(remainder) - 1], s = alpha_len_s[length(alpha_len_s) - 1], t = a_t[length(a_t) - 1]))
}

File: Julia/ExtendedEuclidean.jl
function ExtendedEuclidean(alpha_len, a)
    remainder = [alpha_len, a]
    alpha_len_s = [1, 0]
    a_t = [0, 1]
    quotient = []
    while remainder[end] > 0
        push!(quotient, fld(remainder[end - 1], remainder[end]))
        push!(remainder, mod(remainder[end - 1], remainder[end]))
        push!(alpha_len_s, alpha_len_s[end - 1] - quotient[end] * alpha_len_s[end])
        push!(a_t, a_t[end - 1] - quotient[end] * a_t[end])
    end
    return Dict("gcd" => remainder[end - 1], 
        "s" => alpha_len_s[end - 1], 
        "t" => a_t[end - 1])
end